public/js/common/react_modal.js

```diff
@@ -0,0 +1,84 @@
+function openReactModal(courseCode) {
+    
+    var courseName = getCourseTitle(courseCode);
+    var formattedName = formatCourseName(courseCode);
+    var courseVideoUrls = getCourseVideoUrls(formattedName);
+    
+    React.render(
+        <ModalContent course={formattedName[0]}/>,
+        document.getElementById('modal-content-container')
+    );
+}
+
+var ModalContent = React.createClass({
+    render: function() {
+        return (
+            <div>
+                <Description course={this.props.course} />
+            </div>
+        );
+    }
+});
+
+var Description = React.createClass({
+  getInitialState: function() {
+    return {
+        course: [],
+        sessions: []
+    };
+  },
+
+  componentDidMount: function(){
+      //This loads the course json
+      $.ajax({
+        url: 'course',
+        data: {name: this.props.course},
+        dataType: 'json',
+        success: function(data) {
+            if (this.isMounted()) {
+                this.setState({course: data});
+                
+                console.log(data);
+                
+                //Not sure if there is a cleaner way to do this
+                //This is getting the session times
+                var lectures = data.fallSession.lectures
+                                         .concat(data.springSession.lectures)
+                                         .concat(data.yearSession.lectures);
+                var sessions = [];
+                $.each(lectures, function( index, value ) {
+                    sessions.push(value.code+value.session + "-" + value.section +": " + value.timeStr);
+                    console.log( value.code+value.session + "-" + value.section +": " + value.timeStr );
+                });
+                this.setState({sessions: sessions});
+                console.log("this.state.course.videoUrls", this.state.course.videoUrls);
+            }
+        }.bind(this),
+        error: function(xhr, status, err) {
+            console.error('course-info', status, err.toString());
+        }.bind(this)
+    });
+  },
+    
+  render: function() {
+    //TODO: add video URLs but need to ask David how it works
```

Comment:
`TODO` statements are tricky when merged into master. I think that this is likely something that can be explained in the pull request description, rather than laying here in hopes of not being forgotten.

I, or any other observer, might be able to help you with any questions that you have.